### PR TITLE
Restart weblogin flow if auth failed

### LINKED ISSUE #1373: Can't login with email username
### Actual behaviour
Can't login with email user name. After enter login/password a redirect page appears, but nothing happens more. In server log appears:

{"reqId":"j44SliGSMFW8XhGmIkWM","level":2,"time":"2017-08-18T09:50:54+00:00","remoteAddr":"X.X.X.X","user":"--","app":"core","method":"HEAD","url":"\/remote.php\/webdav\/","message":"Login failed: 'user%40domain.com' (Remote IP: 'X.X.X.X')","userAgent":"Mozilla\/5.0 (Android) ownCloud-android\/2.0.0RC2","version":"12.0.2.0"}
 
### Environment data
Android version: 7.0NRD90M
Device model: Xiaomi Mi5, Elephone p8000
Stock or customized system: Stock system
Nextcloud app version: Nextcloud 2.0RC2
Nextcloud server version: Nextcloud 12.0.2

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Otherwise we have an endless spinning loading dialog.
Discovered while debugging #1373: the existenceCheck failed and thus the loading dialog was not hidden.

Update: Cause of #1373 was the missing proper decoding of username/password which we get from new login flow.

Resolves #1373

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
👍 fine by me

**Comment by AndyScherzinger** [comment]
👍 fine by me

**Comment by AndyScherzinger** [comment]
cc @mario for approval :)

**Comment by mario** [comment]
@AndyScherzinger I did say I want this completed before I approve? (Aka the 60 seconds thing)

**Comment by AndyScherzinger** [comment]
@mario ah, that issue. Sorry, did forget that part 👍

File: src/main/java/com/owncloud/android/authentication/AuthenticatorActivity.java [hunk 1]
```diff
@@ -419,22 +412,37 @@ public void onReceivedError(WebView view, int errorCode, String description, Str
...
-//                return null;
-//            }
-//        }
-//    }
+        // show snackbar after 60s to switch back to old login method
+        new Handler().postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                Snackbar.make(mLoginWebView, "Go back to old login method", Snackbar.LENGTH_INDEFINITE)
+                        .setAction("BACK", new View.OnClickListener() {
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/authentication/AuthenticatorActivity.java]: please put back into the strings.xml

Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/authentication/AuthenticatorActivity.java]: please also put "Go back to old login method" into strings.xml and reference it via ResId

**Comment by AndyScherzinger** [comment]
👍 I'm fine with the approach. What do you think @mario? The "timeout" is now 60 seconds, doesn't redirect you back to the "old" login screen but displays the Snackbar with the message and the back navigation. This way the user can still just to wait if they want to.

**Comment by tobiasKaminsky** [comment]
@mario can you review this? After that we can do a new RC5, or?

**Comment by mario** [comment]
I'd change the string, but nothing urgent for now. Feel free to merge and release RC5.

👍

**Comment by tobiasKaminsky** [comment]
@mario Regarding releasing RC5:  you said to me that in my RC3 release a file for push-notifications was missing. Is this still valid after your & andys fix? if so where is this file?

**Pull request merged**